Version change itself broke `openssh`:
```
       > configure: error: *** zlib too old - check config.log ***
       > Your reported zlib version has known security problems.  It's possible your
       > vendor has fixed these problems without changing the version number.  If you
       > are sure this is the case, you can disable the check by running
       > "./configure --without-zlib-version-check".
       > If you are in doubt, upgrade zlib to version 1.2.3 or greater.
       > See http://www.gzip.org/zlib/ for details.
       For full logs, run 'nix log /nix/store/r2bivy4mdyrzxjs76k7pxim50a0xi69d-openssh-9.4p1.drv'.
```

--- a/pkgs/development/libraries/zlib/default.nix
+++ b/pkgs/development/libraries/zlib/default.nix
@@ -24,7 +24,7 @@ assert splitStaticOutput -> static;
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "zlib";
-  version = "1.2.13";
+  version = "1.3";
 
   src = let
     inherit (finalAttrs) version;
@@ -35,7 +35,7 @@ stdenv.mkDerivation (finalAttrs: {
       # Stable archive path, but captcha can be encountered, causing hash mismatch.
       "https://www.zlib.net/fossils/zlib-${version}.tar.gz"
     ];
-    hash = "sha256-s6JN6XqP28g1uYMxaVAQMLiXcDG8tUs7OsE3QPhGqzA=";
+    hash = "sha256-/wukwpIBPbwnUws6geH5qBPNOd4Byl4Pi/NVcC76WT4=";
   };
 
   postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''

